src/app/code/community/AvS/FastSimpleImport/Model/Import/Entity/Product.php

```diff
@@ -1612,17 +1612,26 @@ protected function _getUploader()
     }
 
     /**
-     * @param $filePath
+     * @param $fileName
      * @return bool
      */
     protected function moveDownloadableFile($fileName)
     {
-        $filePath = $this->_getUploader()->getTmpDir() . $fileName;
-        $destDir = Mage::getConfig()->getOptions()->getMediaDir() . '/downloadable/files/links' . $fileName;
-
+        $filePath      = $this->_getUploader()->getTmpDir() . $fileName;
+        $basePath      = Mage::getModel('downloadable/link')->getBasePath();
+        $destDirectory = dirname(Mage::helper('downloadable/file')->getFilePath($basePath, $fileName));
+        // make sure that the destination directory exists!
+        $ioObject = new Varien_Io_File();
+        try {
+            $ioObject->open(array('path' => $destDirectory));
+        } catch (Exception $e) {
+            $ioObject->mkdir($destDirectory, 0777, true);
```

Comment:
Outch. Thats a bad one but to keep most compatibility I think it is still not OK but acceptable.